micro-rdk/src/common/grpc.rs

```diff
@@ -1435,35 +1439,99 @@ impl<'a> GrpcServerInner<'a> {
+    }
+
+    fn signaling_service_call(
+        self,
+        message: &[u8],
+    ) -> Pin<Box<dyn futures_lite::Stream<Item = Result<Bytes, ServerError>> + Sync + Send>> {
+        let (sender, receiver) = async_channel::bounded::<Result<CallResponse, ServerError>>(1);
+
+        match self.signaling_server {
+            Some(ss) => {
+                let cr = proto::rpc::webrtc::v1::CallRequest::decode(message);
+                let ss = ss.clone();
+                ss.executor
+                    .clone()
+                    .spawn(async move {
+                        match cr {
+                            Ok(cr) => match ss.call(cr, sender.clone()).await {
+                                Ok(()) => {
+                                    sender.close();
+                                }
+                                Err(e) => {
+                                    let _ = sender.send(Err(e)).await;
+                                    sender.close();
+                                }
+                            },
+                            Err(_) => {
+                                let _ = sender
+                                    .send(Err(ServerError::from(GrpcError::RpcInvalidArgument)))
+                                    .await;
+                                sender.close();
+                            }
+                        }
+                    })
+                    .detach();
+            }
+            None => {
+                let _ = sender.send_blocking(Err(ServerError::from(GrpcError::RpcUnimplemented)));
+                sender.close();
+            }
+        }
+
+        Box::pin(receiver.map(move |result| match result {
+            Ok(cr) => GrpcServerInner::encode_message_into_new_buffer(cr),
+            Err(e) => Err(e),
+        }))
+    }
+
+    fn signaling_service_call_update(&mut self, message: &[u8]) -> Result<Bytes, ServerError> {
+        let req = proto::rpc::webrtc::v1::CallUpdateRequest::decode(message)
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
+        let result = match self.signaling_server {
+            Some(ref ss) => ss.call_update(req),
+            None => Err(ServerError::from(GrpcError::RpcUnimplemented)),
+        }?;
+        self.encode_message(result)
+    }
 }
 
 impl<R> WebRtcGrpcService for GrpcServer<R>
 where
     R: GrpcResponse + 'static,
 {
     fn unary_rpc(&mut self, method: &str, data: &Bytes) -> Result<Bytes, ServerError> {
-        {
-            RefCell::borrow_mut(&self.buffer).reserve(GRPC_BUFFER_SIZE);
-        }
-        let mut grpc = GrpcServerInner {
-            buffer: &self.buffer,
+        let grpc = GrpcServerInner {
+            buffer: Some(BytesMut::with_capacity(GRPC_BUFFER_SIZE)),
             robot: &self.robot,
+            signaling_server: &self.signaling_server,
         };
-        grpc.handle_request(method, data)
+        grpc.handle_unary_request(method, data)
```

Comment:
I did not extend streaming support into WebRTC yet, so this can and should bypass the stream based response path and go directly to the result returning path.